server/src/main/java/io/unitycatalog/server/service/Scim2UserService.java

```diff
@@ -0,0 +1,204 @@
+import com.unboundid.scim2.common.exceptions.ScimException;
+import com.unboundid.scim2.common.filters.Filter;
+import com.unboundid.scim2.common.types.Email;
+import com.unboundid.scim2.common.types.Meta;
+import com.unboundid.scim2.common.types.Photo;
+import com.unboundid.scim2.common.types.UserResource;
+import com.unboundid.scim2.common.utils.FilterEvaluator;
+import com.unboundid.scim2.common.utils.Parser;
+import io.unitycatalog.control.model.CreateUser;
+import io.unitycatalog.control.model.UpdateUser;
+import io.unitycatalog.control.model.User;
+import io.unitycatalog.server.exception.BaseException;
+import io.unitycatalog.server.exception.ErrorCode;
+import io.unitycatalog.server.exception.GlobalExceptionHandler;
+import io.unitycatalog.server.exception.Scim2RuntimeException;
+import io.unitycatalog.server.persist.UserRepository;
+import java.net.URI;
+import java.util.Calendar;
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * SCIM2-compliant user management.
+ *
+ * <p>This will be a SCIM 2.0 compliant user management service. The UC User data model will be a
+ * minimal set of required fields that are necessary to support user management/exchange.
+ *
+ * <ul>
+ *   <li>id - internal unique identifier for user
+ *   <li>name - maps to SCIM displayName
+ *   <li>email - maps to SCIM primary email
+ *   <li>externalId - maps to SCIM external id
+ *   <li>userName - maps to SCIM primary email
+ * </ul>
+ */
+@ExceptionHandler(GlobalExceptionHandler.class)
+public class Scim2UserService {
+  private static final UserRepository USER_REPOSITORY = UserRepository.getInstance();
+
+  public Scim2UserService() {}
+
+  @Get("")
+  public HttpResponse getScimUsers(
+      @Param("filter") Optional<String> filter,
+      @Param("startIndex") Optional<Integer> startIndex,
+      @Param("count") Optional<Integer> count) {
+    final Filter userFilter = filter.filter(f -> !f.isEmpty()).map(this::parseFilter).orElse(null);
+    FilterEvaluator filterEvaluator = new FilterEvaluator();
+
+    List<UserResource> userResourcesList =
+        USER_REPOSITORY.listUsers().stream()
+            .map(this::asUserResource)
+            .filter(m -> match(filterEvaluator, userFilter, m))
+            .skip(startIndex.orElse(1) - 1)
+            .limit(count.orElse(50))
+            .toList();
+    return HttpResponse.ofJson(userResourcesList);
+  }
+
+  @Post("")
+  public HttpResponse createScimUser(UserResource userResource) {
+    // Get primary email address
+    Email primaryEmail =
+        userResource.getEmails().stream()
+            .filter(Email::getPrimary)
+            .findFirst()
+            .orElseThrow(
+                () ->
+                    new Scim2RuntimeException(
+                        new PreconditionFailedException("User does not have a primary email.")));
+
+    try {
+      User user = USER_REPOSITORY.getUserByEmail(primaryEmail.getValue());
+      return HttpResponse.ofJson(asUserResource(user));
```

Comment:
Does the SCIM create API do nothing if the user already exists?

Reply:
Is this because deleting a user only sets the user to DISABLED and doesn't delete the DB entry? If so, shouldn't re-creating the user verify that the user's input values match and enable the user?

Reply:
I think this is incorrect behavior (@xx-db) looks lke it should really return a 409 Conflict - https://www.rfc-editor.org/rfc/rfc7644#section-3.3 
> If the service provider determines that the creation of the requested resource conflicts with existing resources (e.g., a "User" resource with a duplicate "userName"), the service provider MUST return HTTP status code 409

Reply:
Done.